wandb/sdk/wandb_run.py

```diff
@@ -1133,9 +1163,17 @@ def _config_callback(
             self._backend.interface.publish_config(key=key, val=val, data=data)
 
     def _config_artifact_callback(
-        self, key: str, val: Union[str, Artifact]
+        self, key: str, val: Union[str, Artifact, dict]
     ) -> Union[Artifact, public.Artifact]:
-        if _is_artifact_string(val):
+        # artifacts can look like dicts as they are passed into the run config
+        # since the run config stores them on the backend as a dict with fields shown
+        # in wandb.util.artifact_to_json
+        if _is_artifact_dict(val):
+            assert isinstance(val, dict)
+            public_api = self._public_api()
+            artifact = public.Artifact.from_id(val["id"], public_api.client)
+            return self.use_artifact(artifact, use_as=key)
+        elif _is_artifact_string(val):
```

Comment:
nit: i can't comment on the else branch at the bottom, but it might be nice to do a type check on the last branch, then throw if it does not meet any of the allowed flavors

Reply:
nice, will also make `_is_artifact` more explicit indicating it is an object `_is_artifact_object`